src/vue/pages/Trade.vue

```diff
@@ -0,0 +1,265 @@
+        :values="formattedPairs"
+        :key="selectedPair"
+        :form-free="true"
+        class="trade-asset-selector__field"
+      />
+    </div>
+    <div class="trade-asset-selector__balances">
+      <p class="trade-asset-selector__balances-value">
+        {{
+          { value: balances.baseBalance, currency: assets.base } | formatMoney
+        }}
+        /
+        {{
+          {value: balances.quoteBalance, currency: assets.quote} | formatMoney
+        }}
+      </p>
+      <p class="trade-asset-selector__balances-label">
+        {{ 'trade.user-balances-label' | globalize }}
+      </p>
+    </div>
+
+    <router-view :component-config="childRoutesConfig" />
+
+    <drawer :is-shown.sync="isCreateBuyOrderDrawerShown">
+      <template slot="heading">
+        {{ 'trade.create-buy-order-form-title' | globalize }}
+      </template>
+      <create-trade-order-form
+        :assets="assets"
+        @close-drawer="closeBuyOrderDrawer"
+      />
+    </drawer>
+    <drawer :is-shown.sync="isCreateSellOrderDrawerShown">
+      <template slot="heading">
+        {{ 'trade.create-sell-order-form-title' | globalize }}
+      </template>
+      <create-trade-order-form
+        :assets="assets"
+        :is-buy="false"
+        @close-drawer="closeSellOrderDrawer"
+      />
+    </drawer>
+  </div>
+</template>
+
+<script>
+import SelectField from '@/vue/fields/SelectField'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+import { Sdk } from '@/sdk'
+import { mapActions, mapGetters } from 'vuex'
+import { vuexTypes } from '@/vuex'
+import CreateTradeOrderForm from '@/vue/forms/CreateTradeOrderForm'
+import Drawer from '@/vue/common/Drawer'
+import TopBar from '@/vue/common/TopBar'
+import { vueRoutes } from '@/vue-router/routes'
+import { Bus } from '@/js/helpers/event-bus'
+import { errors } from '@tokend/js-sdk'
+import { globalize } from '@/vue/filters/globalize'
+
+export default {
+  name: 'trade',
+  components: {
+    SelectField,
+    CreateTradeOrderForm,
+    Drawer,
+    TopBar,
+  },
+  data: () => ({
+    assets: {
+      base: '',
+      quote: '',
+    },
+    balances: {
+      baseBalance: '',
+      quoteBalance: '',
+    },
+    errors,
+    selectedPair: '',
+    tradeablePairs: [],
+    formattedPairs: [],
+    isCreateBuyOrderDrawerShown: false,
+    isCreateSellOrderDrawerShown: false,
+    vueRoutes,
+    childRoutesConfig: {
+      isNeededToReloadData: false,
+      assets: {
+        base: '',
+        quote: '',
+      },
+    },
+  }),
+  computed: {
+    ...mapGetters([
+      vuexTypes.accountBalances,
+    ]),
+  },
+  watch: {
+    selectedPair (value) {
+      const baseAsset = value.split('/')[0]
+      const quoteAsset = value.split('/')[1]
+      this.assets.base = baseAsset
+      this.assets.quote = quoteAsset
+      // pass assets to child router-view components
+      this.childRoutesConfig.assets.base = baseAsset
+      this.childRoutesConfig.assets.quote = quoteAsset
```

Comment:
we bring restrictions this way. I’d recommend extracting quote and base assets on every page of trading feature instead of passing them from a parent

Reply:
Current implementation implies that we take data in one root place and then pass it to other places by a single way. If we will take data from the `route.query` in every single component don't we get a higher probability of error? Also in the current implementation, I can not see any restriction – it's just a `Parent => Child` data passing

Reply:
dicussed IRL, please extract from query params on the needed pages